nipype/interfaces/ants/segmentation.py

```diff
@@ -795,3 +795,107 @@ def _list_outputs(self):
         outputs['output_label_image'] = os.path.abspath(
             self.inputs.output_label_image)
         return outputs
+
+
+class DenoiseImageInputSpec(ANTSCommandInputSpec):
+    dimension = traits.Enum(2, 3, 4, argstr='-d %d', usedefault=False,
+                            desc='This option forces the image to be treated '
+                                 'as a specified-dimensional image. If not '
+                                 'specified, the program tries to infer the '
+                                 'dimensionality from the input image.')
+    input_image = File(exists=True, argstr="-i %s", mandatory=True,
+                       desc='A scalar image is expected as input for noise correction.')
+    noise_model = traits.Enum('Gaussian', 'Rician', argstr='-n %s', usedefault=True,
+                              desc=('Employ a Rician or Gaussian noise model.'))
+    shrink_factor = traits.Int(default_value=1, usedefault=True, argstr='-s %s',
+                               desc=('Running noise correction on large images can '
+                                     'be time consuming. To lessen computation time, '
+                                     'the input image can be resampled. The shrink '
+                                     'factor, specified as a single integer, describes '
+                                     'this resampling. Shrink factor = 1 is the default.'))
+    output_image = traits.Str(argstr="-o %s", genfile=True, hash_files=False,
```

Comment:
you can skip `genfile` (and the need to define `_gen_filename`) if you use `name_source` and `name_template` (more info here: http://nipy.org/nipype/devel/cmd_interface_devel.html#creating-outputs-on-the-fly). This will simplify the interface.